### PR TITLE
N beacons 1 Grp: PR-1

### PR DESCRIPTION
**Pull request by Aadeesh11**
Task completed:
- Update existing models,
    - [x]  Beacon model:
        - [x]  New field,  ```grpID = id of the group belongs to```.
        - [x]  leader is not same as the group’s leader. Here leader is the one who created it, which can later be changed using ```changeLeader()``` mutation.
    - [x]  User Model:
        - [x]  New field, ```groups = [ids of all the groups the user is a part of]```.
- New mutations for creating, joining and fetching a group (changes to be done is resolvers and typedefs).
    - [x]  ```createAGroup()``` ,  fairly straightforward, creating a new group with all the fields and saving it in the DB.
    - [x]  ```joinAGroup(shortcode)```,  Add the new member to all the beacons in the group, also use pubsub to publish this change.
- Update the existing mutations, queries and subscriptions to incorporate group model.
    - [x]  Update ```me``` query.

### CODE REVIEW

File: graphql/resolvers.js [hunk 1]
```diff
@@ -115,6 +116,20 @@ const resolvers = {
...
         },
 
+        createGroup: async (_, { group }, { user }) => {
+            console.log(group);
+            const groupDoc = new Group({
+                leader: user.id,
+                shortcode: nanoid(),
+                ...group,
+            });
+            const newGroup = await groupDoc.save().then(g => g.populate("leader"));
```
Comment (aadibajpai) [inline graphql/resolvers.js:126]: so one minor thing that would be good to handle is the super minor chance that nanoid creates a shortcode that already exists which would result in an error.

File: models/group.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const mongoose = require("mongoose");
+
+const { Schema, model } = mongoose;
+
+const groupSchema = new Schema(
+    {
+        title: String,
+        shortcode: { type: String, required: true },
```
Comment (aadibajpai) [inline models/group.js:8]: this should be unique as well

File: graphql/resolvers.js (hunk 1 above)
Comment (Aadeesh11) [reply graphql/resolvers.js:126]: So, how do we handle that?

Comment (aadibajpai) [reply graphql/resolvers.js:126]: you can see the error that will be thrown and try catch it, and then either retry the thing bc collision twice is very improbable and if that doesn't work either just throw an error that can be handled frontend side and they can implement some retry logic there

Comment (Aadeesh11) [reply graphql/resolvers.js:126]: We would have to check if the current shortcode is used in any other beacon, or something else?

**Comment by Aadeesh11** [comment]
Closing since #167 is with same changes but clean commits !

**Pull request closed without merging**